### PR TITLE
Minor tweaks taken from OpenBSD port of deadbeef

### PR DESCRIPTION
**Pull request by zmyrgel**
Here's small tweaks taken from OpenBSD's port of deadbeef.

### CODE REVIEW

File: plugins.c [hunk 1]
```diff
@@ -1327,7 +1327,7 @@ plug_select_output (void) {
 #else
     char outplugname[100];
 #if defined( HAVE_COCOAUI)
-    conf_get_str ("output_plugin", "CoreAudio", outplugname, sizeof (outplugname));
+    conf_get_str ("output_plugin", "CoreAudio output plugin", outplugname, sizeof (outplugname));
```
Comment (Alexey-Yakovenko) [inline plugins.c]: This is wrong -- the name of the plugin is `CoreAudio`, not `CoreAudio output plugin`

File: plugins/mp3/mp3.h [hunk 2]
```diff
@@ -109,23 +109,16 @@ typedef struct {
     DB_fileinfo_t info;
     // input buffer, for MPEG data
     buffer_t buffer;
-    union {
```
Comment (Alexey-Yakovenko) [inline plugins/mp3/mp3.h]: There _is_ the need to use the union -- it saves some memory.

Perhaps the better idea would be to convert to named unions/structs, instead of this change.

File: Makefile.am [hunk 3]
```diff
@@ -46,12 +46,15 @@ deadbeef_SOURCES =\
...
 	sort.c sort.h\
 	logger.c logger.h
-	
+
 #	ConvertUTF/ConvertUTF.c ConvertUTF/ConvertUTF.h
 
 sdkdir = $(pkgincludedir)
 sdk_HEADERS = deadbeef.h
 
+# Fix 'undefined symbol' errors when loading plugins
```
Comment (Alexey-Yakovenko) [inline Makefile.am]: Neither the commit message, nor this comment, explain what this change really does.

Why are you getting the undefined symbol errors without this change in the first place?

File: plugins.c [hunk 4]
```diff
@@ -1372,8 +1374,10 @@ plug_reinit_sound (void) {
 
     if (plug_select_output () < 0) {
         char outplugname[100];
-#ifdef HAVE_COCOAUI
-        conf_get_str ("output_plugin", "core audio output plugin", outplugname, sizeof (outplugname));
+#if defined(HAVE_COCOAUI)
+        conf_get_str ("output_plugin", "CoreAudio output plugin", outplugname, sizeof (outplugname));
```
Comment (Alexey-Yakovenko) [inline plugins.c]: The wrong coreaudio plugin name again, and the same below.

File: plugins/aac/aac.c [hunk 5]
```diff
@@ -29,7 +29,7 @@
 #include "../../deadbeef.h"
 #include "aac_parser.h"
 
-#include "mp4ff.h"
+#include "../libmp4ff/mp4ff.h"
```
Comment (Alexey-Yakovenko) [inline plugins/aac/aac.c]: It seems that the same fix is needed in ALAC, converter, artwork and artwork-legacy plugins.

File: plugins/gtkui/prefwin.c [hunk 6]
```diff
@@ -187,7 +187,13 @@ gtkui_run_preferences_dlg (void) {
     // output plugin selection
     combobox = GTK_COMBO_BOX (lookup_widget (w, "pref_output_plugin"));
 
+#if defined(HAVE_COCOAUI)
```
Comment (Alexey-Yakovenko) [inline plugins/gtkui/prefwin.c]: This change is in the GTKUI, this condition is always FALSE -- CocoaUI and CoreAudio are on OSX, we don't use GTK there.

We don't need to have coreaudio here.

File: plugins/gtkui/prefwin.c [hunk 7]
```diff
@@ -452,7 +458,13 @@ on_pref_output_plugin_changed          (GtkComboBox     *combobox,
     DB_output_t *new = NULL;
 
     deadbeef->conf_lock ();
+#if defined(HAVE_COCOAUI)
```
Comment (Alexey-Yakovenko) [inline plugins/gtkui/prefwin.c]: same -- cocoaui and coreaudio don't co-exist with GTKUI.

File: conf.c [hunk 8]
```diff
@@ -73,6 +73,7 @@ conf_free (void) {
     }
     conf_items = NULL;
     changed = 0;
+    mutex_unlock (mutex);
```
Comment (Alexey-Yakovenko) [inline conf.c]: This change seems to be wrong. It may cause a race condition, if another thread locks the mutex between mutex_unlock and mutex_free calls. In this case, mutex_free will return `EBUSY` failure, and cause a resource leak.

Why is this change needed?

File: plugins/aac/aac.c (hunk 5 above)
Comment (Alexey-Yakovenko) [inline plugins/aac/aac.c]: Cross this. I'm wrong. Other plugins already do the right thing: `-I@top_srcdir@/plugins/libmp4ff` in Makefile.am

AAC plugin should be doing the same.

Comment (Alexey-Yakovenko) [inline plugins/aac/aac.c]: Err.. but AAC plugin _is_ doing this as well.. Now, I need an explanation why this change is needed on OpenBSD. Everything should work without it.

File: plugins.c (hunk 1 above)
Comment (zmyrgel) [inline plugins.c]: Ok, I'll provide more detailed commit messages, need to learn a bit of git-fu first.
I'll sync the CoreAudio naming, seems easy.

File: plugins/mp3/mp3.h (hunk 2 above)
Comment (zmyrgel) [inline plugins/mp3/mp3.h]: Or perhaps this change can stay local in OpenBSD ports, OpenBSD is still using gcc 4.2.1 but recently imported llvm to base so the default compiler at least on some arches might change in the near future.

File: Makefile.am (hunk 3 above)
Comment (zmyrgel) [inline Makefile.am]: Hmm, I'm not actually sure. Its in the OpenBSD's patches and tried to dig old commit logs and mailing lists on why these were done but couldn't find it. 
My guess is that it has something to do with archs still using GCC3.
But I'll verify it.

File: plugins/gtkui/prefwin.c (hunk 6 above)
Comment (zmyrgel) [inline plugins/gtkui/prefwin.c]: Ok, good to know.

File: plugins/aac/aac.c (hunk 5 above)
Comment (zmyrgel) [inline plugins/aac/aac.c]: Ok, I'll work on this a bit more. These all might be workarounds for older GCC versions ( GCC2 was still used by some arches a while back).

**Comment by Alexey-Yakovenko** [comment]
I took the commits aa86494e1f57339a53748dacbc38791dd25a4b9d and d93fd99b74966c138b4ddf582802fe2b91d6f72c from this PR. The rest is doing more harm than good, so I won't take them without explanation and significant rework.
